frontend/src/editor/transcription_editor.tsx

```diff
@@ -70,32 +75,54 @@ function renderElement({ element, children, attributes }: RenderElementProps): J
   throw Error('Unknown element type');
 }
 
-function renderLeaf({ leaf, children, attributes }: RenderLeafProps): JSX.Element {
-  const classes = ['word'];
-  if (leaf.start !== undefined) {
-    classes.push(startTimeToClassName(leaf.start));
-  }
+const Leaf = memo(
+  ({
+    start,
+    conf,
+    children,
+    attributes,
+  }: {
+    start?: number;
+    conf?: number;
+    children: RenderLeafProps['children'];
+    attributes: RenderLeafProps['attributes'];
+  }): JSX.Element => {
+    const systemPrefersDark = useMediaQuery('(prefers-color-scheme: dark)');
```

Comment:
does it help performance to optimize this and don't do this once per word / batch it in useMediaQuery?

Reply:
A bit, about to 2ms